devlib/trace/perf.py

```diff
@@ -108,7 +124,23 @@ def stop(self):
 
     # pylint: disable=arguments-differ
     def get_trace(self, outdir):
-        for label in self.labels:
+        for label, report_opt in zip(self.labels, self.report_optionstrings):
+            #in record mode generate report and copy
+            if self.perf_mode == 'record':
+                # .rpt
+                command = self._build_perf_report_command(report_opt, label)
+                self.target.execute(command, as_root=True)
+                target_file = self._get_target_reportfile(label)
+                host_relpath = os.path.basename(target_file)
+                host_file = _f(os.path.join(outdir, host_relpath))
+                self.target.pull(target_file, host_file)
+                # .trace
+                target_file = self._get_target_tracefile(label)
+                host_relpath = os.path.basename(target_file)
+                host_file = _f(os.path.join(outdir, host_relpath))
+                target_file = self._get_target_perfdatafile()
+                self.target.pull(target_file, host_file)
```

Comment:
This might be the cause of the permission errors: `perf record` seems to generate a `perf.data` file with permissions `600` (owned by `root` as it's running through `su`) so that `self.target.pull` (which ends up doing an `adb pull` which is seen, from the target filesystem, as an access from a non-`root` user) would fail. The solution might be to use `self.target.pull(..., as_root=True)`?